### DISCUSSION TITLE
How to get the current most outer parent node type in focus?

### DISCUSSION THREAD

**Discussion by katerlouis** [ueberdosis/tiptap#3787] [General Questions] [answered]
I'm desparately trying to get the current node type in a function triggered by a menu button (where the user can toggle headlines, bold etc.)

I made a custom paragraph with different variants; I did the same with bulletList and orderedList. All of these extensions also got a `size` attribute, which can only be `small`, `medium` or `large` – simple stuff. And because all sizes are the same, I'd like to simplify this function.

I need to get `size` of the current block node; but `getAttributes()` requires a typeName to check for. But How can I get the current nodeType (or typeName) without explicitly checking for every possible node beforehand? This is very repeatitive code.

`isActive()` seems to know what node has been in focus before the user clicked the button. Why isn't there a function getting me the curernt node / type name? 

Here's what **I'd like to have**
```js
toggleSize(size) {
  const currentTypeName = getCurrentNode();
  const currentSize = editor.getAttributes(currentTypeName).size;

  // toggle back to default size
  if (currentSize && currentSize === size) { ... }
  
  // set desired size
  else { ... }
}
```

Here's what I have:
```js
  toggleSize(size) {
      const { editor } = this;

      if (editor.isActive('bulletList')) {
        const currentSize = editor.getAttributes('bulletList').size;
        const newSize = currentSize === size
          ? 'medium' // default size
          : size
        ;
        editor.chain().focus().updateAttributes('bulletList', { size: newSize }).run();
      }
      
      if (editor.isActive('orderedList')) {
        editor.chain().focus().updateAttributes('orderedList', { size }).run();
      }
    },

  // do the same for a few more node types

```

EDIT:
I found a way to get the node, but unfortunately not the most outer one, the actual block I'm working in: `editor.view.state.selection.$anchor.node();` gives me the paragraph node that is inside my listItem of bulletList, although I want to get `bulletList`

**Comment by DhaniAM** [comment, marked as answer]
I found the way to get the parent node from current selection:
```js
const state = editor.view.state;
const from = state.selection.from;
const pos = state.doc.resolve(from);
const parent = pos.node(pos.depth - 1);
```

**Reply by TPLCarloTaleon** [reply]
Cool so, this is essentially how to get the "current" node as well.

```ts
const state = editor.view.state;
const from = state.selection.from;
const pos = state.doc.resolve(from);

pos.node() // current node
```
